crates/ruff_server/src/server.rs

```diff
@@ -160,18 +114,69 @@ impl Server {
                     api::request(req)
                 }
                 lsp::Message::Notification(notification) => api::notification(notification),
-                lsp::Message::Response(response) => {
-                    tracing::error!(
-                        "Expected request or notification, got response instead: {response:?}"
-                    );
-                    continue;
-                }
+                lsp::Message::Response(response) => scheduler.response(response),
```

Comment:
I may not have enough context on this piece. Why is this now not an error?

Reply:
It was an error before because we didn't have any reason to receive a response from the server. Now that we have an interface to send the client requests, we do want to listen for responses.